zipline/testing/fixtures.py

```diff
@@ -292,3 +299,126 @@ def init_class_fixtures(cls):
         start_loc = end_loc - cls.TRADING_DAY_COUNT
 
         cls.trading_days = all_days[start_loc:end_loc + 1]
+
+
+class WithPipelineEventDataLoader(WithAssetFinder):
+    """
+    ZiplineTestCase mixin providing common test methods/behaviors for event
+    data loaders.
+
+    `get_sids` must return the sids being tested.
+    `get_dataset` must return {sid -> pd.DataFrame}
+    `loader_type` must return the loader class to use for loading the dataset
+    `make_asset_finder` returns a default asset finder which can be overridden.
+    """
+    @abc.abstractproperty
+    def get_sids(cls):
+        raise NotImplementedError('get_sids')
+
+    @abc.abstractproperty
+    def get_dataset(self):
+        raise NotImplementedError('get_dataset')
+
+    @abc.abstractproperty
+    def loader_type(self):
+        raise NotImplementedError('loader_type')
+
+    @classmethod
+    def make_asset_finder(cls):
+        return cls.enter_class_context(
+            tmp_asset_finder(
+                equities=make_simple_equity_info(
+                    cls.get_sids(),
+                    start_date=pd.Timestamp('2013-01-01', tz='UTC'),
+                    end_date=pd.Timestamp('2015-01-01', tz='UTC'),
+                )
+            )
+        )
+
+    def loader_args(self, dates):
+        """Construct the base  object to pass to the loader.
+
+        Parameters
+        ----------
+        dates : pd.DatetimeIndex
+            The dates we can serve.
+
+        Returns
+        -------
+        args : tuple[any]
+            The arguments to forward to the loader positionally.
+        """
+        return dates, self.get_dataset()
+
+    def setup_engine(self, dates):
```

Comment:
we may want to namespace these methods like: pipeline_event_loader_setup_engine` because we expect to have potentially many fixtures that all provide data. This make the consumer better understand which fixture they are hooking.